Assignment5/JiangWenqi-21A272/assignment_5.py

```diff
@@ -0,0 +1,305 @@
+ck.configure_kgtk(input_graph_path=input_path,
+                  output_path=output_path,
+                  project_name=project_name)
+
+# Commented out IPython magic to ensure Python compatibility.
+# %%time
+# kgtk("""
+#     query -i all
+#         --match '(instance)-[:P31]->()'
+#         --return 'count(distinct instance) as count_instances'
+# """)
+
+"""Now, count the number of distinct properties: 
+
+"""
+
+# Commented out IPython magic to ensure Python compatibility.
+# %%time
+# kgtk("""
+#     query -i all
+#         --match '()-[p]->()'
+#         --return 'count(distinct p.label) as count_properties'
+# """)
+
+"""Now, let's count the frequency of those properties. That is, how many instances we can find with each property"""
+
+# Commented out IPython magic to ensure Python compatibility.
+# %%time
+# kgtk("""
+#     query -i all
+#         --match '(i)-[p]->()'
+#         --return 'p.label, count(distinct i) as count_properties'
+#         --order-by 'cast(count_properties,int) desc'
+#         / add-labels
+# """)
+
+"""## Simple queries
+Some of these queries are simple and will run in the Wikidata endpoint. 
+Try both of them using SPARQL and Kypher
+"""
+
+# Which actors has Schwarzenegger worked with throughout his career? (Print also the movie)
+
+# (in SPARQL)
+# TO DO 
+query = """
+SELECT ?colleage ?colleageLabel ?film ?filmLabel
+WHERE
+{
+  ?film wdt:P161 wd:Q2685.
+  ?film wdt:P161 ?colleage.
+  FILTER(?colleage != wd:Q2685)
+  SERVICE wikibase:label { bd:serviceParam wikibase:language "en". }
+}
+"""
+# In Kypher:
+kgtk("""
+  query -i all
+    --match '(colleage)<-[:P161]-(film)-[:P161]->(:Q2685)'
+    --where 'colleage != "Q2685"'
+    --return 'colleage, film'
+    / add-labels
+""")
+
+# How many awards does Schwarzenegger have?
+
+# SPARQL:
+# TO DO
+query= """
+SELECT (COUNT(?award) AS ?num_awards)
+WHERE 
+{
+  wd:Q2685 wdt:P166 ?award
+}
+"""
+
+# Kypher:
+kgtk("""
+    query -i all
+        --match '(:Q2685)-[:P166]->(award)'
+        --return 'count(award) as num_awards'
+""")
+
+# Retrieve at least two members of Schwarzenegger's political party. Make sure only persons are returned
+# SPARQL:
+# TO DO
+query = """SELECT ?member ?political_party
+WHERE
+{
+  wd:Q2685 wdt:P102 ?political_party.
+  ?member wdt:P102 ?political_party.
+  ?member wdt:P31 wd:Q5.
+  FILTER (?member != wd:Q2685)
+}
+"""
+# Kypher:
+kgtk("""
+     query -i all
+        --match '(:Q2685)-[:P102]->(political_party)<-[:P102]-(member)-[:P31]->(:Q5)'
+        --where 'member != "Q2685"'
+        --return 'member, political_party'
+    / add-labels   
+""")
+
+# What are the properties that describe an artist?
+
+# In theory this one is heavy on Wikidata
+
+# SPARQL: 
+query = """SELECT DISTINCT ?property
+WHERE
+{
+  ?class rdfs:subClassOf* wd:Q483501.
+  ?subject wdt:P106 ?class.
+  ?subject ?property ?p.
+}
+"""
+
+# In Kypher:
+kgtk("""
+    query -i all
+        --match '()<-[l {label: property}]-()-[:P279*]->(:Q483501)'
```

Comment:
Not correct. You are printing the properties of the class instead of the elements belonging to that profession

Reply:
Thank you for correcting! Happy new year, Darijo.

Reply:
Happy new year! The right solution is something like `()<-[l {label: property}]-()-[:P106]->()-[:P279*]->(:Q483501)`